### PR TITLE
Cleanup Control event args

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs [hunk 1]
```diff
@@ -2,198 +2,138 @@
...
+        public Color BackColor
+        {
+            get
+            {
+                if ((State & DrawItemState.Selected) == DrawItemState.Selected)
+                {
                     return SystemColors.Highlight;
                 }
-                return backColor;
+
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: nit: remove extra space

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs [hunk 2]
```diff
@@ -2,198 +2,138 @@
...
-                return state;
+        public Color ForeColor
+        {
+            get
+            {
+                if ((State & DrawItemState.Selected) == DrawItemState.Selected)
+                {
+                    return SystemColors.HighlightText;
+                }
+
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: nit: remove extra line

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs [hunk 3]
```diff
@@ -2,198 +2,138 @@
...
-        ///     Draws a handy focus rect in the given rectangle.
+        /// Draws a handy focus rect in the given rectangle.
         /// </devdoc>
-        public virtual void DrawFocusRectangle() {
-            if ((state & DrawItemState.Focus) == DrawItemState.Focus
-                && (state & DrawItemState.NoFocusRect) != DrawItemState.NoFocusRect)
+        public virtual void DrawFocusRectangle()
+        {
+            if ((State & DrawItemState.Focus) == DrawItemState.Focus
+                && (State & DrawItemState.NoFocusRect) != DrawItemState.NoFocusRect)
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: nit: can you put the && at the end of the previous line? I realize this is not how the file was but I think it's a little clearer

File: src/System.Windows.Forms/src/System/Windows/Forms/KeyEventArgs.cs [hunk 4]
```diff
@@ -2,189 +2,96 @@
...
+                Keys keyGenerated =  KeyData & Keys.KeyCode;
 
                 // since Keys can be discontiguous, keeping Enum.IsDefined.
                 if (!Enum.IsDefined(typeof(Keys),(int)keyGenerated))
+                {
                     return Keys.None;
-                else
-                    return keyGenerated;
+                }
+
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/KeyEventArgs.cs]: nit: remove extra space

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs [hunk 5]
```diff
@@ -2,141 +2,57 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-namespace System.Windows.Forms {
+using System.Runtime.InteropServices;
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs]: can you move this back under the namespace?

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs [hunk 6]
```diff
@@ -2,141 +2,57 @@
...
-            this.x = x;
-            this.y = y;
-            this.delta = delta;
+    [ComVisible(true)]
+    public class MouseEventArgs : EventArgs
+    {
+        /// <devdoc>
+        /// Initializes a new instance of the <see cref='System.Windows.Forms.MouseEventArgs'/> class.
+        /// </devdoc>
+        public MouseEventArgs(MouseButtons button, int clicks, int x, int y, int delta)
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs]: Would you mind writing a few constructor unit tests for these?

File: src/System.Windows.Forms/src/System/Windows/Forms/PaintEventArgs.cs [hunk 7]
```diff
@@ -59,125 +51,105 @@ internal static string GetAllocationStack()
...
+            {
+                if (_graphics == null && _dc != IntPtr.Zero)
+                {
+                    oldPal = Control.SetUpPalette(_dc, force: false, realizePalette: false);
+                    _graphics = Graphics.FromHdcInternal(_dc);
+                    _graphics.PageUnit = GraphicsUnit.Pixel;
+                    _savedGraphicsState = _graphics.Save(); // See ResetGraphics() below
                 }
-                return graphics;
+
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/PaintEventArgs.cs]: nit: remove extra line

File: src/System.Windows.Forms/src/System/Windows/Forms/PaintEventArgs.cs [hunk 8]
```diff
@@ -59,125 +51,105 @@ internal static string GetAllocationStack()
...
-        ///       rectangle in which to paint.
-        ///    </para>
+        /// Gets the rectangle in which to paint.
         /// </devdoc>
-        public Rectangle ClipRectangle {
-            get {
-                return clipRect;
-            }
-        }
+        public Rectangle ClipRectangle { get; set; }
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/PaintEventArgs.cs]: **no set;**

File: src/System.Windows.Forms/src/System/Windows/Forms/DpiChangedEventArgs.cs [hunk 9]
```diff
@@ -2,52 +2,33 @@
...
         }
 
-        /// <include file='doc\DpiChangedEventArgs.uex' path='docs/doc[@for="DpiChangedEventArgs.DeviceDpiXOld"]/*' />
-        /// <devdoc>
-        /// </devdoc>
-        public int DeviceDpiOld {
-            get;
-            private set;
-        }
+        public int DeviceDpiOld { get; }
```
[the remark this answers is not in the document]
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DpiChangedEventArgs.cs]: yes, it was unused

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs (hunk 6 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs]: We have these already

File: src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs (hunk 5 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/MouseEventArgs.cs]: Coding guidelines are that usings at top of file, out of namespace

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs (hunk 2 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: IMO space makes things clearer, but updated this

**Comment by hughbe** [comment]
Fixed PR feedback. Just saying that I don't plan on adding ANY public API or modifying ANY behaviour in my PRs. I've accidentally pushed changes that might do so in the past but none of these are deliberate - i noticed the team's reviewed these as if I wanted to make deliberate changes in the past, but no, the public API is fixed and I get that it can't be changed :)

File: src/System.Windows.Forms/src/System/Windows/Forms/PaintEventArgs.cs (hunk 7 above)
Comment (sharwell) [reply src/System.Windows.Forms/src/System/Windows/Forms/PaintEventArgs.cs]: Really? I've always seen the opposite request.

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs (hunk 3 above)
Comment (sharwell) [reply src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: Ick ☹️

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs (hunk 2 above)
Comment (karelz) [reply src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: I agree with @hughbe. I believe that @sharwell had same comment later on.
I am not sure if this is captured in coding style in CoreFX in written form, but I'd recommend at minimum to get to consensus in WinForms team if you want to deviate from "common practice" @zsd4yr ;)

**Comment by zsd4yr** [comment]
> the public API is fixed and I get that it can't be changed :)

This is not quite true; any proposed api changes simply need to go through review! Very few things are set-in-stone so to speak

File: src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs (hunk 2 above)
Comment (zsd4yr) [reply src/System.Windows.Forms/src/System/Windows/Forms/DrawItemEventArgs.cs]: I'm not married to either way tbh; I just thought it looked a little cleaner without

**Pull request merged**